tests/integration/testlib/vxlan.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+import libnmstate
+
+from libnmstate.schema import VXLAN
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+
+VXLAN_IFNAME = '{}.{}'
+
+
+def vxlan_desired_state(vxlans):
+    vxlans_up = []
+    for vxlan in vxlans:
+        vxlans_up.append(vxlan_up(*vxlan))
+    return {Interface.KEY: vxlans_up}
+
+
+@contextmanager
+def vxlan_interfaces(vxlans, extra_interfaces=[]):
```

Comment:
In python you cannot use mutable default values.
But it is also not that nice to ask this function to control other interfaces (extra ones).

Try to reach a design where each function/contextmanager does one thing and then just combine them together here or in the test that calls it to combine them.
For example, helpers create the state, then the test merges them and apply it. we do things like that with other tests.